include/mapnik/map.hpp

```diff
@@ -277,11 +271,6 @@ class MAPNIK_DECL Map : boost::equality_comparable<Map>
      */
     std::vector<layer> const& layers() const;
 
-    /*! \brief Get all layers.
-     *  @return Non-constant layers.
-     */
-    std::vector<layer> & layers();
-
```

Comment:
Removing this method breaks `demo/viewer` as it makes at least two calls to this, at least one of which is actually trying to modify the layer list...

Reply:
You're right, removing non-const access to layers is problematic in viewer (same goes for Python bindings). I'm going to look into this again. Thanks for highlighting this issue.